From a666516fa3bc6e0d6b11d715ed014f97110386d8 Mon Sep 17 00:00:00 2001
From: Arien Judge <arienjudge@outlook.com>
Date: Fri, 13 Oct 2023 18:42:45 +1100
Subject: [PATCH] toolchain: link include directory when using external
 toolchain

Some packages, eg perl, explicitly reference directories in
$(TOOLCHAIN_DIR), which, for an external toolchain, don't currently exist.
This commit adds a target for external toolchains to symlink the include
directory to support other packages which may be doing this.

Signed-off-by: Arien Judge <arienjudge@outlook.com>
---
 toolchain/Makefile | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/toolchain/Makefile b/toolchain/Makefile
index 09c16f72a7808..dfb8880980d23 100644
--- a/toolchain/Makefile
+++ b/toolchain/Makefile
@@ -76,9 +76,22 @@ $(TOOLCHAIN_DIR)/info.mk $(STAGING_DIR)/.prepared: $(TOOLCHAIN_DIR)/stamp/.ver_c
 endif
 endif
 
+ifeq ($(CONFIG_EXTERNAL_TOOLCHAIN),)
+  $(TOOLCHAIN_DIR)/include: ;
+else
+  $(TOOLCHAIN_DIR)/include: .config;
+	@for dir in $(TOOLCHAIN_DIR); do ( \
+		$(if $(QUIET),,set -x;) \
+		mkdir -p "$$dir"; \
+		cd "$$dir"; \
+		ln -nsf $(TOOLCHAIN_ROOT_DIR)/include include; \
+	); done
+	@touch $@
+endif
+
 # prerequisites for the individual targets
 $(curdir)/ := .config prereq
-$(curdir)//compile = $(STAGING_DIR)/.prepared $(TOOLCHAIN_DIR)/info.mk $(tools/stamp-compile)
+$(curdir)//compile = $(STAGING_DIR)/.prepared $(TOOLCHAIN_DIR)/info.mk $(TOOLCHAIN_DIR)/include $(tools/stamp-compile)
 
 ifndef DUMP_TARGET_DB
 $(TOOLCHAIN_DIR)/stamp/.gcc_final_installed:
